File: NewsApp/app/src/main/java/com/example/android/topic41/data/database/CacheInterface.java
package com.example.android.topic41.data.database;

import android.os.AsyncTask;

import com.example.android.topic41.domain.util.Article;

import java.util.List;

import io.reactivex.Single;

public interface CacheInterface {
    Single<List<Article>> getArticlesByTheme(String theme);

    void refresh(List<Article> articles, Long currentTime);
}

File: NewsApp/app/src/main/java/com/example/android/topic41/data/database/CacheInterface.kt
package com.example.android.topic41.data.database

import com.example.android.topic41.domain.util.Article
import io.reactivex.Single

interface CacheInterface {
    fun getArticlesByTheme(theme: String): Single<List<Article>>
    fun refresh(articles: List<Article>, currentTime: Long)
}
